src/database.ts

```diff
@@ -723,6 +749,26 @@ class Database extends common.GrpcServiceObject {
     });
   }
 
+  public _metadataWithRequestId(
+    nthRequest: number,
+    attempt: number,
+    priorMetadata?: {[k: string]: string}
+  ): {[k: string]: string} {
+    if (!priorMetadata) {
+      priorMetadata = {};
+    }
+    const withReqId = {
+      ...priorMetadata,
+    };
+    withReqId[X_GOOG_SPANNER_REQUEST_ID_HEADER] = craftRequestId(
+      this._clientId || 1,
+      1, // TODO: Properly infer the channelId
```

Comment:
@olavloite I filed a TODO for retrieving the channelId which I believe can be done much later in a phased series of PRs instead of at once.